sklearn/decomposition/dict_learning.py

```diff
@@ -729,6 +744,12 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
 
     if method not in ('lars', 'cd'):
         raise ValueError('Coding method not supported as a fit algorithm.')
+
+    if method == 'lars' and positive_code:
+        raise ValueError(
+            "Positive constraint not supported for 'lars' coding method."
+        )
```

Comment:
rather than duplicating this error message can you make a private function. Something like

`_check_positive_coding(method, positive)`

that would crash if needed.